### PR TITLE
Suppress viewer error messages in sendMessage

### PR DESCRIPTION
**Pull request by muxin**
Follow up of https://github.com/ampproject/amphtml/pull/6525
Fixes https://github.com/ampproject/amphtml/issues/6526

### CODE REVIEW

File: src/service/viewer-impl.js [hunk 1]
```diff
@@ -1012,8 +1012,13 @@ export class Viewer {
           eventType, data, awaitResponse));
     }
 
+    // If there is no viewer, no further action is needed.
```
Comment (jridgewell) [inline src/service/viewer-impl.js]: This comment only applies to the `.resolve` case.

File: src/service/viewer-impl.js [hunk 2]
```diff
@@ -1012,8 +1012,13 @@ export class Viewer {
...
     if (!this.messagingReadyPromise_) {
-      return Promise.reject(getChannelError());
+      if (awaitResponse) {
+        return Promise.reject(getChannelError());
+      } else {
+        return Promise.resolve();
```
Comment (jridgewell) [inline src/service/viewer-impl.js]: Actually, this is not equivalent to the code we had before https://github.com/ampproject/amphtml/pull/6508. To get back to that, we need something like:

```js
if (awaitPromise && !this.messagingReadyPromise_) {
  return Promise.reject(getChannelError());
}
```

Comment (muxin) [inline src/service/viewer-impl.js]: But if there's no viewer, there's no need to do the action below like adding it into the queue.

Comment (lannka) [inline src/service/viewer-impl.js]: :-) this is what i did here: https://github.com/ampproject/amphtml/pull/6525/files

But after discussion, we think it does not make sense to queue up the message if there will ever be a messaging chanel

Comment (jridgewell) [inline src/service/viewer-impl.js]: Apologies, I read it as `messageDeliverer` and not `messagingReadyPromise`.

**Pull request merged**